pyteal/ast/abi/tuple.py

```diff
@@ -523,20 +523,37 @@ def __init__(self):
         if not anns:
             raise Exception("Expected fields to be declared but found none")
 
-        self.type_specs: OrderedDict[str, TypeSpec] = OrderedDict()
+        # NOTE: this `_ready` variable enables `__setattr__` during `__init__` execution,
+        # while after `__init__`, we cannot use `__setattr__` to set fields in `NamedTuple`.
+        # The reason for `_ready` is that, the field naming scheme in Python would change
+        # _double underscore led field name_ to something else,
+        # while _single underscore led variable names_ would stay intact.
+        self._ready = False
```

Comment:
nit: `__ready` would be more consistent with the other field names

Reply:
tho `__ready` is consistent with other field's name, it unfortunately cannot fit here, for python would swap the naming of attributes, see: https://stackoverflow.com/questions/1301346/what-is-the-meaning-of-single-and-double-underscore-before-an-object-name, so we need to deliberately change the `__getattr__` implementation

Reply:
> the __getattr__ method would need to check if name == "_NamedTuple__ready"

yes exactly, then it is non trivial to see why `__ready` gets changed into `_NamedTuple__ready`, for the change is implicit.

Reply:
Ok great, this is the last thing blocking an approval for me

Reply:
I added a few comments above declaration of `self._ready`, hopefully that makes sense to you.

Reply:
Sorry if I was unclear, my preference is to rename `_ready` to `__ready`.

The reason being we should minimize the number of field names that cannot be used by `NamedTuple` subclasses.

Reply:
ugh ok. I will swap the naming then.

Reply:
alright, final pass for a naming change...